src/Widgets/Alarm.vala

```diff
@@ -81,11 +81,20 @@ public class Hourglass.Widgets.Alarm : Gtk.ListBoxRow {
 
     private string make_date_label () {
         var comp = new GLib.DateTime.now_local ();
-        if (Granite.DateTime.is_same_day (time, comp)) {
+        int today = comp.get_day_of_week () != 7 ? comp.get_day_of_week () : 0;
+        if (
+            Granite.DateTime.is_same_day (time, comp) && repeat.length == 0 ||
+            today in repeat
+        ) {
             return _("Today");
-        } else {
-            return Granite.DateTime.get_relative_datetime (time);
         }
+
+        if (repeat.length > 0) {
+            int[] next_repeat = {repeat[0]};
+            return Utils.selected_days_to_string (next_repeat);
```

Comment:
```suggestion
            return Utils.selected_days_to_string (repeat);
```

I thinked about the bug you mentioned in the first comment; doesn't simply passing the integer array `repeat` to this method work?